scripts/generate-global-changelog.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const {execSync} = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+const IGNORE_LIST = [
+	'packages/demos/CHANGELOG.md',
+	'clayui.com/CHANGELOG.md',
+	'CHANGELOG.md',
+];
+
+const dateObj = new Date();
+const month = dateObj.getMonth() + 1;
+const day = dateObj.getDate();
+const year = dateObj.getFullYear();
+
+const dateString = `${year}-${month}-${day}`;
+
+const globalChangeLog = path.join(__dirname, '../CHANGELOG.md');
+
+function main() {
+	const diffOutput = execSync(
+		'git diff HEAD^ --name-only -- "*CHANGELOG.md"'
+	).toString();
+
+	const changelogContents = fs.readFileSync(globalChangeLog).toString();
+
+	if (changelogContents.includes(`## ${dateString}`)) {
+		console.error(
+			`CHANGELOG has already been generated for the ${dateString} release`
+		);
+
+		return;
+	}
+
+	const files = diffOutput
+		.split(/\n/)
+		.filter((file) => !IGNORE_LIST.includes(file))
+		.filter(Boolean);
+
+	let output = `# CHANGELOG\n## ${dateString}\n`;
+
+	for (let i = 0; i < files.length; i++) {
+		const file = files[i];
+
+		try {
+			const fileDiff = execSync(`git diff HEAD^ -U0 -- "${file}"`)
+				.toString()
+				.split(/\n@@.*/)[1]
+				.trim()
+				.replace(/(\n|^)\+/g, '\n');
```

Comment:
One of the problems with `git diff` is that there are a _lot_ of settings that people can put in their `~/.gitconfig` that can change the output (see the references to `diff.` in `man git-config`), so it would probably be more robust to just read the file contents.

```js
const VERSION_HEADING = /^#+ \[\d/;

function getChanges(file) {
    const changes = [];
    
    for (const line of fs.readFileSync(file, 'utf8').split('\n')) {
        if (VERSION_HEADING.test(line)) {
            if (changes.length) {
                break;
            }
            else {
                changes.push(line);
            }
        }
    }

    return changes.join('\n');
}
```

(Untested, written in browser, so may have bugs.)